packages/@aws-cdk/aws-redshift-alpha/lib/private/database-query-provider/table.ts

```diff
@@ -208,6 +209,12 @@ async function updateTable(
 
   await Promise.all(alterationStatements.map(statement => executeStatement(statement, tableAndClusterProps)));
 
+  const oldTableNamePrefix = oldResourceProperties.tableName.prefix;
+  if (tableNamePrefix !== oldTableNamePrefix) {
+    await executeStatement(`ALTER TABLE ${tableName} RENAME TO ${tableNamePrefix + tableNameSuffix}`, tableAndClusterProps);
+    return tableNamePrefix + tableNameSuffix;
+  }
+
```

Comment:
This looks like the crux of the change. Altering the name of the table instead of creating a new table. Is there a behavioral difference of having this function at the end of the update method rather than the beginning like the create call?

Reply:
Yes, so the existing code can continue to operate. Given that this creates a new `tableName`, the old code will still be using the old `tableName`. Therefore the changes to table will fail. This is even more so apparent now, given the edit you wanted regarding the `newTableName` variable, which makes it even more apparent to leave it as it is.